spec/spec_helper.rb

```diff
@@ -1,3 +1,33 @@
 # encoding: utf-8
 require "logstash/devutils/rspec/spec_helper"
 require 'logstash/inputs/unix'
+
+class UnixSocketHelper
+
+  attr_reader :path
+
+  def initialize
+    @socket = nil
+  end
+
+  def new_socket(path)
+    @path   = path
+    File.unlink if File.exists?(path) && File.socket?(path)
+    @socket = UNIXServer.new(path)
+    self
+  end
+
+  def loop(forever=false)
+    @thread = Thread.new do
+      s = @socket.accept
+      s.puts "hi" while forever
+    end
+    self
+  end
+
+  def close
+    @thread.kill
+    @socket.close
+    File.unlink(path)
+  end
+end
```

Comment:
I agree with your reasoning, I am just trying to have a standard way of where the helpers should be located.


Also, this will work since the test file is explicitely doing `require_relative "../spec_helper"` which isn't the case with all the plugins.